src/stream.rs

```diff
@@ -407,27 +410,38 @@ pub(crate) fn connect_host(
         stream.set_write_timeout(unit.agent.config.timeout_write)?;
     }
 
-    if proto == Some(Proto::HTTPConnect) {
+    if proto == Some(Proto::HTTPConnect) || proto == Some(Proto::HTTPSConnect) {
         if let Some(ref proxy) = proxy {
-            write!(stream, "{}", proxy.connect(hostname, port)).unwrap();
-            stream.flush()?;
+            let mut s;
```

Comment:
I'd love to have a more descriptive variable name here, and a type annotation. With the current code, the reader has to read _forward_ in the code to figure out what type `s` is.